s3.go

```diff
@@ -293,9 +293,7 @@ func (b *s3Batch) Commit() error {
 	}
 
 	for _, k := range putKeys {
-		jobs <- func() error {
-			return b.s.Put(k, b.ops[k.String()].val)
-		}
+		jobs <- b.newPutJob(k, b.ops[k.String()].val)
```

Comment:
Ok, we _need_ tests for this package.

Reply:
We're experimenting with a private cluster, so I'll find time to do that!

Reply:
That would be *awesome*. Is there any way to test s3 with mocks? My blocker here has always been a lack of access to some S3 bucket.

Reply:
At Docker we used to test against a dockerized minio server, so we could try that. An alternative would be write a in-memory version for the AWS SDK s3 interface: https://github.com/aws/aws-sdk-go/blob/master/service/s3/s3iface/interface.go#L62